src/sdk/object_sdk.js

```diff
@@ -94,6 +108,38 @@ class ObjectSDK {
         return this._setup_bucket_namespace(bucket);
     }
 
+    // TODO: Handle accountspace operations / RPC auth (i.e system, account, anonymous) and anonymous access
+    async authorize_request_account({ bucket, op_name }) {
+        const ns = bucket && await this.read_bucket_sdk_namespace_info(bucket);
+        const token = this.get_auth_token();
+        if (!token) return;
+        if (ns) return 'NotImplemented';
+        let account;
+        try {
+            account = await account_cache.get_with_cache({
+                rpc_client: this.internal_rpc_client,
+                access_key: token.access_key
+            });
+        } catch (error) {
+            dbg.error('authorize_request_account error:', error);
+            return 'InvalidAccessKeyId';
+        }
+        this._has_account_bucket_permission({ account, bucket, token });
+    }
+
+    _has_account_bucket_permission({ account, bucket, token }) {
+        const signature = signature_utils.get_signature_from_auth_token(token, account.access_keys[0].secret_key.unwrap());
```

Comment:
Why is this signature check here in the bucket permission check and not in the `authorize_request_account` function?